manifests/pool.pp

```diff
@@ -71,8 +71,8 @@
   Optional[String[1]] $gateway              = undef,
   Optional[Variant[Array[String[1],1],String[1]]] $range = undef,
   Optional[String[1]] $failover             = undef,
-  Optional[String[1]] $options              = undef,
-  Optional[String[1]] $parameters           = undef,
+  Optional[Array[String]] $options          = undef,
```

Comment:
This is a backwards incompatible change. If you go that route, you can also choose to always initialize the array.
```suggestion
  Array[String[1]] $options          = [],
```

Then in the template you can drop `<% if @options.is_a? Array -%>`.

Reply:
@ekohl Thank you for quick answer. What do you mean "backwards incompatible change"? is that it will fail on installations where "options" is defined as String?
The reason I propose these changes is that after update from v4.0.1 to v5.0.0 it fails with "options" definead as list in YAML file
As I understood from documentation "options" is always (if defined) a list, each as a separate line in config, there are no syntax to put all options in one line as one string

Reply:
> What do you mean "backwards incompatible change"? is that it will fail on installations where "options" is defined as String?

Correct, exactly that. A compatible data type (that only fixes the regression) would be `Variant[Undef, String, Array[String]]`. In that case the template shouldn't be modified. That could be released as 5.0.1.

Reply:
Thanks, I've updated PR